src/Uno.UI.Runtime.Skia.Gtk/Input/GtkCorePointerInputSource.cs

```diff
@@ -596,4 +610,100 @@ private void RaisePointerWheelChanged(PointerEventArgs ptArgs, [CallerMemberName
 
 		PointerWheelChanged?.Invoke(this, ptArgs);
 	}
+
+	/// <summary>
+	/// The coordinates received in the event args are relative to the widget that raised the event.
+	/// Usually, that's the native overlay (which embodies the entire window, minus the titlebar). However,
+	/// if another overlay (e.g. TextBox) raises an event, we still want the coordinates relative to the
+	/// top-left of the window. This method makes that adjustment
+	/// </summary>
+	private static void AdjustCoords(object o, EventCrossing e)
```

Comment:
There is a lot of repetition. Can you refactor that?

Reply:
Yeah, but the lines of code to redactor will be like 5 lines shorter or something, so it barely matters.

Reply:
@ramezgerges I mean `AdjustCoords` is not very small and is duplicated for each event type. Could it be centralized somehow? e.g, move that logic as a single call in `AsPointerArgs(Event)`?